### ISSUE TITLE
Revert MSYS2 compatibility changes.

### ISSUE THREAD

**Issue by DragoonX6** [mesonbuild/meson#252]
I happened to use the wrong MSYS2 packages when trying out meson.
We ended up writing some hacks to support MSYS2, this is in hindsight not needed.
In fact, the MSYS2 changes are now breaking proper builds.
MSYS2 is comparable to cygwin (it's a fork actually) in how it does stuff, but it also comes with toolchains targeting windows by default.
Where did it go wrong? I ended up using MSYS2 python instead of the python package that targets windows. This messed up platform detection and the paths output in the build files.
In short, I should've installed mingw-w64-x86_64-python3 package, the one that actually reports Windows as platform, then there wouldn't have been any problems.

**Comment by jpakkane** [comment]
Which part of the change is causing breakages and how is it happening?

**Comment by DragoonX6** [comment]
Mainly the detect_pythonbin function, this causes it to call the wrong python binary every time ninja needs to regenerate.
Reverting to sys.executable will make it work on both MSYS2 python and Windows python.
Given you have compiled ninja with the MSYS compatibility layer, MSYS2 python will work. You need the earlier mentioned python package to get it to work as "windows", to make native builds.
